### PR TITLE
Document SAP system aggregate

### PR DESCRIPTION
**Pull request by arbulu89**
Document SAP system aggregate code. I have included some additional documentation in the SAP application instance registering command, to explain how the application and database are associated.
This part of the code is a bit hidden, but I hope the new docs help a bit on that.

PD: We cannot put docs in the protocol code as it is a `defimpl`, that's why it goes in the register command, which at the end looks an appropriate place

### CODE REVIEW

File: lib/trento/domain/sap_system/sap_system.ex [hunk 1]
```diff
@@ -1,5 +1,44 @@
...
+
+  In order to have a fully registered SAP system, both the database and application
+  composing this system must be registered. And each of the two layers might be composed
+  by multiple instances altogether. This means that a SAP system aggregate stream can have
+  multiple application/database instances.
+
+  ## SAP instance
+
+  A SAP instance can be understood as a single SAP workload installation running in a
+  particular host. So the instance runs entirely in one host, but in the other hand
```
Comment (dottorblaster) [inline lib/trento/domain/sap_system/sap_system.ex:15]:
```suggestion
  particular host. So the instance runs entirely in one host, but on the other hand
```

File: lib/trento/domain/sap_system/sap_system.ex [hunk 2]
```diff
@@ -1,5 +1,44 @@
 defmodule Trento.Domain.SapSystem do
-  @moduledoc false
+  @moduledoc """
+  The SAP system aggregate manages all the domain logic related to
+  deployed SAP systems, which are composed by a database and application layers.
+  **The HANA database is the unique supported database type.**
```
Comment (dottorblaster) [inline lib/trento/domain/sap_system/sap_system.ex:5]:
```suggestion
  **The HANA database is the only supported database type.**
```

File: lib/trento/domain/sap_system/sap_system.ex [hunk 3]
```diff
@@ -1,5 +1,44 @@
...
+
+  In order to have a fully registered SAP system, both the database and application
+  composing this system must be registered. And each of the two layers might be composed
+  by multiple instances altogether. This means that a SAP system aggregate stream can have
+  multiple application/database instances.
+
+  ## SAP instance
+
+  A SAP instance can be understood as a single SAP workload installation running in a
```
Comment (dottorblaster) [inline lib/trento/domain/sap_system/sap_system.ex:14]:
```suggestion
  A SAP instance can be seen as a single SAP workload installation running in a
```

File: lib/trento/domain/sap_system/sap_system.ex [hunk 4]
```diff
@@ -1,5 +1,44 @@
...
+  particular host. So the instance runs entirely in one host, but in the other hand
+  multiple different SAP instances might be running in the same host.
+
+  For example, a HANA database might be composed by two database instances
+  that are working together in a System replication scenario.
```
Comment (dottorblaster) [inline lib/trento/domain/sap_system/sap_system.ex:19]:
```suggestion
  that are working together in a System Replication scenario.
```

File: lib/trento/domain/sap_system/sap_system.ex [hunk 5]
```diff
@@ -1,5 +1,44 @@
...
+
+  ## SAP system registration process
+
+  The SAP system registration process have some caveats, so let's see them in more details.
```
Comment (dottorblaster) [inline lib/trento/domain/sap_system/sap_system.ex:23]:
```suggestion
  The SAP system registration process has some caveats, so let's see them in more details.
```

**Review by dottorblaster — CHANGES_REQUESTED** [review]
Just some really tiny suggestions, awesome job man

**Comment by arbulu89** [comment]
Thanks @dottorblaster ,
All changed. All were good suggestions!

**Review by dottorblaster — APPROVED** [review]
LGTM

File: lib/trento/domain/sap_system/sap_system.ex [hunk 6]
```diff
@@ -1,5 +1,44 @@
 defmodule Trento.Domain.SapSystem do
-  @moduledoc false
+  @moduledoc """
+  The SAP system aggregate manages all the domain logic related to
+  deployed SAP systems, which are composed by a database and application layers.
+  **The HANA database is the only supported database type.**
+
+  In order to have a fully registered SAP system, both the database and application
+  composing this system must be registered. And each of the two layers might be composed
+  by multiple instances altogether. This means that a SAP system aggregate stream can have
```
Comment (fabriziosestito) [inline lib/trento/domain/sap_system/sap_system.ex:9]: i would use state instead of stream

**Review by fabriziosestito — CHANGES_REQUESTED** [review]
Great work! Just a nitpick and then go!

File: lib/trento/domain/sap_system/sap_system.ex (hunk 6 above)
Comment (arbulu89) [reply lib/trento/domain/sap_system/sap_system.ex]: done!

**Review by nelsonkopliku — APPROVED** [review]
Gonna read this many times :stuck_out_tongue: 
Good job!

**Pull request merged**